mne/source_space.py

```diff
@@ -2535,36 +2615,37 @@ def get_volume_labels_from_aseg(mgz_fname, return_colors=False):
     label_colors : list of str
         The RGB colors of the labels included in this mgz file.
 
+    See Also
+    --------
+    read_freesurfer_lut
+
     Notes
     -----
+    .. versionchanged:: 0.21.0
+       The label names are now sorted in the same order as their corresponding
+       values in the MRI file.
     .. versionadded:: 0.9.0
     """
     import nibabel as nib
-
-    # Read the mgz file using nibabel
-    mgz_data = _get_img_fdata(nib.load(mgz_fname))
-
-    # Get the unique label names
-    lut = _get_lut()
-
-    label_names = [lut[lut['id'] == ii]['name'][0]
-                   for ii in np.unique(mgz_data)]
-    label_colors = [[lut[lut['id'] == ii]['R'][0],
-                     lut[lut['id'] == ii]['G'][0],
-                     lut[lut['id'] == ii]['B'][0],
-                     lut[lut['id'] == ii]['A'][0]]
-                    for ii in np.unique(mgz_data)]
-
-    order = np.argsort(label_names)
-    label_names = [label_names[k] for k in order]
-    label_colors = [label_colors[k] for k in order]
-
+    atlas = nib.load(mgz_fname)
+    want = np.unique(_get_img_fdata(atlas))
+    ids, colors = read_freesurfer_lut()
```

Comment:
Would it be possible to add a way to use specific freesurfer LUT here ? Adding an input either an object returned by or a file path to be read by `read_freesurfer_lut` :

`def get_volume_labels_from_aseg(mgz_fname, lut=None, return_colors=False):`

Reply:
So you want both `read_freesurfer_lut(fname=None)`, and `get_volume_labels_from_aseg(mgz_fname, return_colors=False, lut=None)` where non-None inputs mean "don't use the builtin one"? Seems doable

Reply:
Actually `read_freesurfer_lut` already takes `fname`, so we just need `lut` support for `get_volume_labels_from_aseg`

Reply:
added `atlas_ids` argument in latest commit